user.go

```diff
@@ -0,0 +1,186 @@
+package goharborclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-openapi/runtime"
+
+	"github.com/mittwald/goharbor-client/api/v1.10.0/client/products"
+	"github.com/mittwald/goharbor-client/api/v1.10.0/model"
+)
+
+const (
+	ErrUserNotFound     = "user not found on server side"
+	ErrUserBadRequest   = "Unsatisfied with constraints of the user creation/modification."
+	ErrUserMismatch     = "id/name pair not found on server side"
+	ErrUserAlreadyExist = "User with this username already exists."
+	ErrUserInvalidID    = "Invalid user ID."
+)
```

Comment:
Personally, I think lower-casing these and removing any punctuation will look better in the end.
```suggestion
const (
	ErrUserNotFound     = "user not found on server side"
	ErrUserBadRequest   = "unsatisfied with constraints of the user creation/modification"
	ErrUserMismatch     = "id/name pair not found on server side"
	ErrUserAlreadyExist = "user with this username already exists"
	ErrUserInvalidID    = "invalid user ID"
)
```
Maybe add comments to each const definition as well.

Reply:
Fixed